site/src/pages/docs/client-factory-builder.mdx

```diff
@@ -0,0 +1,129 @@
+a single connection(or `EventLoop` which is a thread) can handle, so you might want to increase
+the number of connections(or `EventLoop` threads) for certain or all endpoints.
+
+You can do this via <type://ClientFactoryBuilder#maxNumEventLoopsPerEndpoint(int)>:
+
+The <type://ClientFactoryBuilder> class has a feature to set the number of `EventLoop` threads using
+`maxNumEventLoopsPerEndpoint` and `maxNumEventLoopsFunction`. The former parameter will set the number of `EventLoop`
+threads for each endpoint (This parameter is used for HTTP/2. If you want to articulate this feature in HTTP/1,
+use `maxNumEventLoopsPerHttp1Endpoint` instead.). It indicates that if you change this value `N`,
+then Client instances that are generated by the factory will use distinct `EventLoop` threads per endpoint.
+
+```java
+try (ClientFactory factory =
+        ClientFactory.builder()
+                     // a client will use maximum 5 EventLoops from the worker group per endpoint
+                     .maxNumEventLoopsPerEndpoint(5)
+                     .build()) {
+    // Do something using factory
+}
+
+try (ClientFactory factory =
+        ClientFactory.builder()
+                     // or you can just use Integer.MAX_VALUE to use all EventLoops
+                     .maxNumEventLoopsPerEndpoint(Integer.MAX_VALUE)
+                     .build()) {
+    // Do something using factory
+}
+```
+
+The client might need to send many requests to a certain endpoint, while not bothering other endpoints.
+In that situation, you can use <type://ClientFactoryBuilder#maxNumEventLoopsFunction(ToIntFunction)>:
+
+```java
+try (ClientFactory factory =
+        ClientFactory.builder()
+                     .maxNumEventLoopsFunction(endpoint -> {
+                         if (endpoint.equals(Endpoint.of("lotsOfTraffic.com"))) {
+                         // We are going to use all EventLoops to send reqeusts.
+                             return Integer.MAX_VALUE;
+                         }
+                         // We use just one `EventLoop` per endpoint for the rest.
+                         return 1;
+                     })
+                     .build()) {
+    // Do something using factory
+}
+```
+
+If HTTP/1.1 is used, the client can establish many connections even though only one EventLoop is used.
+It is because of the nature that a client can send only one request at a time using HTTP/1.1 connection.
+You can increase the number of `EventLoop` threads per HTTP/1.1 endpoint to distribute the load
+via <type://ClientFactoryBuilder#maxNumEventLoopsPerHttp1Endpoint(int)>.
+
+## Customizing TLS
+
+You can configure the TLS settings of your client connection
+with <type://ClientFactoryBuilder#tlsCustomizer(Consumer)>. For example, you could specify a different
+CA certificate chain, configure TLS or enforce certain cipher suites.
+
+```java
+final String resourceRoot = "/path/to/resourceRoot";
+final String privateKeyPath = "private/keyPath";
```

Comment:
`resourceRoot + privateKeyPath` will be `"/path/to/resourceRootprivate/keyPath"`.

```
final String resourceRoot = "/path/to/resource/";
```
and inline `privateKeyPath` for consistency?
```java
getClass().getResourceAsStream(resourceRoot + "private.key"))
```